insonmnia/worker/server.go

```diff
@@ -2011,6 +2030,10 @@ func (m *Worker) AskPlanByTaskID(taskID string) (*sonm.AskPlan, error) {
 	return m.salesman.AskPlan(planID)
 }
 
+func (m *Worker) Metrics(ctx context.Context, req *sonm.WorkerMetricsRequest) (*sonm.WorkerMetricsResponse, error) {
+	return m.metrics.Get(), nil
```

Comment:
Seems like there is no way to break collecting if a client interrupts the request. Pass the context.

Reply:
Here `.Get()` is just returning cached value with lock, metrics collection already has the root ctx of the worker, so it will be canceled when a worker will be closed.